### ISSUE TITLE
Offline Tiles from Mapbox not Working

### ISSUE THREAD

**Issue by jasonash** [StraboSpot/strabo-mobile#490]
Saving tiles for offline use from Mapbox satellite or topo results in blank white tiles being served. The same is true of custom maps from Mapbox. OSM tiles seem to be working correctly. This issue exists on both iOS and Android versions.

**Comment by jessicagood** [comment]
@jasonash Neither @NathanNovak can recreate this with fresh installs on Android and iOS.  Downloaded tiles for Mapbox Satellite and Topo seem to be working when offline. Any more info?

**Comment by jasonash** [comment]
@panamint55 Are you able to do a fresh install to test whether if works for you?

**Comment by jessicagood** [comment]
Yea, on 3 different devices (Android and iOS) we uninstalled all versions of StraboSpot and did fresh installs from the stores and it works fine.

**Comment by jasonash** [comment]
@jessicagood  One thing I noticed is that the map is pulling from one URL:

http://api.tiles.mapbox.com/v4/mapbox.satellite/4/4/5.png?access_token=pk.eyJ1Ijoic3RyYWJvLWdlb2xvZ3kiLCJhIjoiY2lpYzdhbzEwMDA1ZnZhbTEzcTV3Z3ZnOSJ9.myyChr6lmmHfP8LYwhH5Sg

While the tile downloader is pulling from a different url:

http://a.tiles.mapbox.com/v4/mapbox.satellite/4/4/5.jpg?access_token=pk.eyJ1Ijoic3RyYWJvLWdlb2xvZ3kiLCJhIjoiY2lpYzdhbzEwMDA1ZnZhbTEzcTV3Z3ZnOSJ9.myyChr6lmmHfP8LYwhH5Sg

The second request is doing some strange redirection in Chrome:
<img width="482" alt="screenshot 2018-05-17 09 52 20" src="https://user-images.githubusercontent.com/7904402/40187874-3f53aec6-59be-11e8-9176-ce5b7a968712.png">

I just did another test, and it still doesn't work for me. Here is a screengrab from the phonegap iOS build:
![img_0039](https://user-images.githubusercontent.com/7904402/40188007-8e31dd9c-59be-11e8-9db5-6b8ef9f0c853.png)

**Comment by jasonash** [comment]
Here is a screenshot from a fresh install of phonegap Android:
![screenshot_20180517-104040](https://user-images.githubusercontent.com/7904402/40188231-19910566-59bf-11e8-9931-69d7a290ff42.png)

**Comment by jessicagood** [comment]
Yes, the urls should be slightly different. I thought the problem was with the store builds? Btw, both the Android store builds and phonegap build say 1.1.0 but iOS Store says 1.1.1.

**Comment by jessicagood** [comment]
Updating the phonegap build right now

**Comment by jasonash** [comment]
Yes, I had to upload a new binary to iTunes because of the camera crashing. I haven’t uploaded a new binary to Android Yet.

**Comment by panamint55** [comment]
Works with fresh install.


***********************************************
J. Douglas Walker
Isotope Geochemistry Laboratories
Nichols Hall
2335 Irving Hill Drive
Lawrence, KS  66045
***********************************************

**Comment by jasonash** [comment]
@jessicagood Here is a video of my complete process:

https://youtu.be/2kka0-QbWoc

**Comment by jasonash** [comment]
@jessicagood Yes, after downloading offline tiles I completely close the app.

**Comment by jasonash** [comment]
@jessicagood Ah, I see the workaround where you can press the map symbol on the other basemaps page. I'll pass this along to the end-users who were reporting the issue just in case this helps them out today. Thanks.

**Comment by jessicagood** [comment]
We can get it working on iOS when you go to any other page actually too actually.

And I thought I was recreating the issue on Android but I just didn't have the correct map selected. So Android still looks fine to me.

**Comment by NathanNovak** [comment]
I think I was able to recreate it on android.

**Comment by jasonash** [comment]
I can duplicate the issue on Android by doing the following:
1. While online, download map for use offline
2. Completely close out Strabo App
3. Turn off all radios on device (wifi and cellular)
4. Start Strabo App, at which time maps are gone (white)

I can then click on the "other basemaps" menu item and click on the map icon to "activate" the map, just like the workaround mentioned above by Jessica for iOS.

**Comment by NathanNovak** [comment]
We rolled back the version to three prior commits: 
7f584a9 (Nov 7 ,2017) ,  6284265 (Sept 26, 2017), and ed4410 (Sept 7, 2017 -v 1.0.0)

The offline map in every version the map did the same thing.  

Jason if you found a version where it was working would you let us know?

**Comment by jasonash** [comment]
@jessicagood I have set up our own tilecache that is ready for testing. The URL is tiles.strabospot.org

So this old URL:
http://b.tiles.mapbox.com/v4/mapbox.satellite/19/119885/201168.png?access_token=pk.eyJ1Ijoic3RyYWJvLWdlb2xvZ3kiLCJhIjoiY2lpYzdhbzEwMDA1ZnZhbTEzcTV3Z3ZnOSJ9.myyChr6lmmHfP8LYwhH5Sg

Becomes:
http://tiles.strabospot.org/v4/mapbox.satellite/19/119885/201168.png?access_token=pk.eyJ1Ijoic3RyYWJvLWdlb2xvZ3kiLCJhIjoiY2lpYzdhbzEwMDA1ZnZhbTEzcTV3Z3ZnOSJ9.myyChr6lmmHfP8LYwhH5Sg

Right now, the only valid layers are mapbox.satellite and mapbox.outdoors.

I did a few tests, and the results look pretty good. Downloading ~3500 tiles had zero stalls and took roughly the same time as directly from Mapbox. Subsequent downloads of the same ~3500 tiles were about an order of magnitude faster than from Mapbox. If this proves successful, we can move forward with a zip download if needed.

**Comment by NathanNovak** [comment]
@jasonash I updated the URL for mapbox.satellite and mapbox.outdoors with commit ea045b6e52ac0d045512dade1f17a9b03b7df89e
